python/lsst/meas/base/applyApCorr.py

```diff
@@ -0,0 +1,156 @@
+#
+# LSST Data Management System
+# Copyright 2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import math
+
+import lsst.pex.config
+import lsst.pex.exceptions
+import lsst.afw.image
+import lsst.pipe.base
+from .base import getApCorrNameSet
+
+__all__ = ("ApplyApCorrConfig", "ApplyApCorrTask")
+
+class ApCorrInfo(object):
+    """!Catalog field names and keys needed to aperture correct a particular flux
+    """
+    def __init__(self, schema, algName):
+        """!Construct an ApCorrInfo and add fields to the schema
+
+        @param[in,out] schema  source catalog schema;
+            three fields are used to generate keys:
+            - {algName}_flux
+            - {algName}_fluxSigma
+            - {algName}_fluxFlag
+            three fields are added:
+            - {algName}_ApCorr
+            - {algName}_ApCorrSigma
+            - {algName}_flag_apcorr
+        @param[in] algName  name of flux measurement algorithm, e.g. "base_PsfFlux"
+        """
+        self.algName = algName  # name of flux measurement algorith
+        self.fluxName = algName + "_flux"           # name of flux field
+        self.fluxSigmaName = algName + "_fluxSigma" # name of flux sigma field
+        self.fluxKey = schema.find(self.fluxName).key           # key to flux field
+        self.fluxSigmaKey = schema.find(self.fluxSigmaName).key # key to flux sigma field
+        self.fluxFlagKey = schema.find(algName + "_flag").key   # key to flux flag field
+        self.apCorrKey = schema.addField(
+            algName + "_ApCorr",
+            doc = "aperture correction applied to %s" % (algName,),
+            type = float,
+        )
+        self.apCorrSigmaKey = schema.addField(
+            algName + "_ApCorr_Sigma",
+            doc = "aperture correction applied to %s" % (algName,),
+            type = float,
+        )
+        self.apCorrFlagKey = schema.addField(
+            algName + "_flag_apcorr",
+            doc = "set if unable to aperture correct %s" % (algName,),
+            type = "Flag",
+        )
```

Comment:
Capitalization and underscores are a bit inconsistent in these last three fields.  I propose:
 - `algName + "_apCorr"`
 - `algName + "_apCorrSigma"`  (note lack of underscore, to match e.g. `"_fluxSigma"`)
 - `algName + "_flag_apCorr"`